packages/manager/apps/web/client/app/domain-operation/operation-table/operation-table.controller.js

```diff
@@ -0,0 +1,133 @@
+import set from 'lodash/set';
+import isEmpty from 'lodash/isEmpty';
+
+export default class OperationTableController {
+  /* @ngInject */
+  constructor($scope, $timeout, Alerter, domainOperationService) {
+    this.$scope = $scope;
+    this.$timeout = $timeout;
+    this.Alerter = Alerter;
+    this.Operation = domainOperationService;
+  }
+
+  $onInit() {
+    this.stepPath = '';
+
+    this.$scope.alerts = { dashboard: 'domains.operations.alerts' };
+    this.$scope.resetAction = () => this.$scope.setAction(false);
+    this.$scope.setAction = (action, data) => {
+      this.$scope.currentAction = action;
+      this.$scope.currentActionData = data;
+
+      if (action) {
+        this.stepPath = `${this.$scope.currentAction}.html`;
+        $('#currentAction').modal({
+          keyboard: true,
+          backdrop: 'static',
+        });
+      } else {
+        $('#currentAction').modal('hide');
+        this.$scope.currentActionData = null;
+        this.$timeout(() => {
+          this.stepPath = '';
+        }, 300);
+      }
+    };
+
+    this.$scope.$on('domains.operations.relaunched', this.getOperationIds);
+    this.$scope.$on('$locationChangeStart', () => {
+      this.$scope.resetAction();
+    });
+
+    this.getOperationIds();
+  }
+
+  transformItem(id) {
+    return this.getOperation(id);
+  }
+
+  static capitalizeFirstLetter(string) {
+    return string.charAt(0).toUpperCase() + string.slice(1);
+  }
+
+  resetSearch() {
+    this.filters[this.constructor.normalizeName(this.type)] = '';
+    this.getOperationIds();
+  }
+
+  getOperation(args) {
+    return this.Operation[
+      `get${this.constructor.capitalizeFirstLetter(this.type)}Operation`
+    ](args);
+  }
+
+  getOperationIds() {
```

Comment:
Maybe you can directly use the result of the get as a binding of this component instead of making a generic function